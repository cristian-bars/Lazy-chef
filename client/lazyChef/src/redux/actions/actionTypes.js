export default {
  LOAD_RECIPES: 'LOAD_RECIPES',
  DELETE_RECIPE: 'DELETE_RECIPE',
  ADD_RECIPE: 'ADD_RECIPE',
  UPDATE_RECIPE: 'UPDATE_RECIPE',
  LOAD_RECIPE: 'LOAD_RECIPE',

  LOAD_USERS: 'LOAD_USERS',
  DELETE_USER: 'DELETE_USER',
  ADD_USER: 'ADD_USER',
  UPDATE_USER: 'UPDATE_USER',
  LOAD_USER: 'LOAD_USER',
};
