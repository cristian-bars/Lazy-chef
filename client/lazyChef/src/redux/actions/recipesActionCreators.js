import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/recipes';
const url = 'http://192.168.0.29:2022/recipes';
//const url = 'http://192.168.1.130:2022/recipes';

export function loadRecipes() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_RECIPES,
        recipes: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPES_ERROR',
      });
    }
  };
}

export function addRecipe(recipe) {
  return {
    type: actionTypes.ADD_RECIPE,
    recipe,
  };
}

export function deleteRecipe(recipeId) {
  return async dispatch => {
    await axios.delete(`${url}/${recipeId.id}`);
    dispatch({
      type: actionTypes.DELETE_RECIPE,
      recipeId,
    });
  };
}

export function updateRecipe(recipe) {
  return {
    type: actionTypes.UPDATE_RECIPE,
    recipe,
  };
}

export function loadRecipe(recipe) {
  return {
    type: actionTypes.LOAD_RECIPE,
    recipe,
  };
}

export function getRecipeById(recipeId) {
  console.log(`${url}/${recipeId}`);
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/${recipeId}`);
      dispatch({
        type: actionTypes.LOAD_RECIPE,
        recipe: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_RECIPE_ERROR',
      });
    }
  };
}
