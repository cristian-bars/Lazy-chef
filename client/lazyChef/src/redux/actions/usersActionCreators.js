import actionTypes from './actionTypes';
import axios from 'axios';

//const url = 'http://localhost/users';
const url = 'http://192.168.0.29:2022';
//const url = process.env.REACT_APP_PORT;

export function loadUsers() {
  return async dispatch => {
    try {
      const {data} = await axios(url);
      dispatch({
        type: actionTypes.LOAD_USERS,
        users: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}

export function addUsers(user) {
  return {
    type: actionTypes.ADD_USER,
    user,
  };
}

export function deleteUser(userId) {
  return async dispatch => {
    await axios.delete(`${url}/${userId.id}`);
    dispatch({
      type: actionTypes.DELETE_USER,
      userId,
    });
  };
}

export function updateUser(user) {
  return {
    type: actionTypes.UPDATE_USER,
    user,
  };
}

export function loadUser(user) {
  return {
    type: actionTypes.LOAD_USER,
    user,
  };
}

export function getUserById(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/login`, user);
      dispatch({
        type: actionTypes.LOAD_USER,
        userToken: data,
      });
    } catch (error) {
      dispatch({
        type: 'LOAD_USERS_ERROR',
      });
    }
  };
}
