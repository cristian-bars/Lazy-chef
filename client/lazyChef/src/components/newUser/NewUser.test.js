import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Store from 'redux-mock-store';
import * as actions from '../../redux/actions/usersActionCreators';
import {Provider} from 'react-redux';
import Register from './index';

describe('Given a Register component', () => {
  const mockStore = Store();
  let myStore;
  let navigation;

  const goBack = jest.fn();

  beforeEach(() => {
    jest.spyOn(actions, 'addUsers').mockReturnValueOnce({type: ''});
    myStore = mockStore({
      user: {name: 'Cristian'},
    });
    navigation = {
      navigate: jest.fn(),
    };
  });
  describe('When is render', () => {
    describe('And user.isLogged is false', () => {
      test('Then is mathced to snapshot', () => {
        const tree = render(
          <Provider store={myStore}>
            <Register navigation={navigation} />
          </Provider>,
        );

        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe('And user have token', () => {
    test('Then navigation is invoked ', () => {
      myStore = mockStore({
        user: {
          name: 'Cristian',
          email: 'cristian@lazychef.com',
        },
      });

      render(
        <Provider store={myStore}>
          <Register navigation={navigation} />
        </Provider>,
      );

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });

  describe('When Register button is pressed', () => {
    test('Then navigation.navigate is invoked', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <Register navigation={navigation} />
        </Provider>,
      );
      const register = getByTestId('register');
      fireEvent.press(register);

      expect(actions.addUsers).toHaveBeenCalled();
    });
  });

  // describe('When back button is pressed', () => {
  //   test('Then navigation.navigate is invoked', () => {
  //     const {getByTestId} = render(
  //       <Provider store={myStore}>
  //         <Register navigation={goBack} />
  //       </Provider>,
  //     );
  //     const goToBack = getByTestId('goBack');
  //     fireEvent.press(goToBack);

  //     expect(navigation.goBack).toHaveBeenCalled();
  //   });
  // });

  // describe('When nameInput is typing with Cristian', () => {
  //   test('Then nameInput value is Cristian', () => {
  //     const {getByTestId} = render(
  //       <Provider store={myStore}>
  //         <Register />
  //       </Provider>,
  //     );
  //     const nameInput = getByTestId('nameInput');
  //     const name = 'Cristian';
  //     fireEvent.changeText(nameInput, name);

  //     expect(nameInput.props.value).toBe(name);
  //   });
  // });

  // describe('When emailInput is typing with cristian@gmail.com', () => {
  //   test('Then emailInput value is cristian@gmail.com', () => {
  //     const {getByTestId} = render(
  //       <Provider store={myStore}>
  //         <Register />
  //       </Provider>,
  //     );
  //     const emailInput = getByTestId('emailInput');
  //     const email = 'cristian@gmail.com';
  //     fireEvent.changeText(emailInput, email);

  //     expect(emailInput.props.value).toBe(email);
  //   });
  // });

  // describe('When passwordInput is typing with 1234ABC', () => {
  //   test('Then passwordInput value is 1234ABC', () => {
  //     const {getByTestId} = render(
  //       <Provider store={myStore}>
  //         <Register />
  //       </Provider>,
  //     );
  //     const passwordInput = getByTestId('passwordInput');
  //     const password = '1234ABC';
  //     fireEvent.changeText(passwordInput, password);

  //     expect(passwordInput.props.value).toBe(password);
  //   });
  // });
});
