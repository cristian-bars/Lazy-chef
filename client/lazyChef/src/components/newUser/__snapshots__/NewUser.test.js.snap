// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Register component When is render And user.isLogged is false Then is mathced to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#044AFD",
      "flex": 1,
      "justifyContent": "center",
      "padding": 24,
      "paddingTop": 50,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 10,
        "left": 20,
        "opacity": 1,
        "padding": 20,
        "position": "absolute",
        "top": 20,
        "width": 10,
        "zIndex": 1,
      }
    }
    testID="goBack"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/img/back.png",
        }
      }
      style={
        Object {
          "height": 20,
          "left": -12,
          "position": "relative",
          "top": -10,
          "width": 20,
        }
      }
    />
  </View>
  <Image
    source={
      Object {
        "uri": "https://i.ibb.co/MVc000Y/Logo-transparent-Patxi.png",
      }
    }
    style={
      Object {
        "alignItems": "flex-start",
        "height": 250,
        "width": 250,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 25,
        "fontWeight": "bold",
        "paddingVertical": 20,
      }
    }
  >
    Nuevo usuario
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Nombre"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    testID="nameInput"
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    testID="emailInput"
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "marginVertical": 5,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    testID="passwordInput"
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 30,
        "opacity": 1,
        "width": 300,
      }
    }
    testID="register"
  >
    <Text
      style={
        Object {
          "fontFamily": "Cochin",
          "fontSize": 25,
        }
      }
    >
      Registrarse
    </Text>
  </View>
</View>
`;
